extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+        # get and parse "lastUpdatedOn" from MongoDB/Alissa for specific patient_dn_no
+        lastUpdatedOn_mongoDB = db.find_one({"dn_no" : patient_dn_no}, {"lastUpdatedOn":1, "_id":0})
+        for key, value in lastUpdatedOn_mongoDB.items():
+            lastUpdatedOn_mongoDB = value
+        lastUpdatedOn_Alissa = ONB01_analysis['lastUpdatedOn']
+        if lastUpdatedOn_Alissa == lastUpdatedOn_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no) 
+            return
+        elif lastUpdatedOn_Alissa > lastUpdatedOn_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif lastUpdatedOn_Alissa < lastUpdatedOn_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+            sys.exit('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+        else:
+            logger.info('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+            sys.exit('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+
+    # get all relevant info from one patient into one dictionary.
+    patient = {}
+    patient["dn_no"] = patient_dn_no
+    logger.info('start uploading to mongodb: %s' % patient_dn_no)
+    patient["patient_accession_no"] = accession_number
+    patient.update(ONB01_analysis)
+    patient["analysis"] = patient.pop("reference")
+    patient["platform_dataset"] = analyis_sources["platformDataSet"]["info"]
+
+    # changes the format of the annotation sources from Alissa, to fit in MongoDB and resemble the old O-schijf format
+    externalSources_dict = {}
+    variants = VUS_response
+    for key in analyis_sources['annotationSources']:
+        source_name = ""
+        source_value = ""
+        for x,y in key.items():
+            if x == 'name':
+                source_name = y
+            elif x == 'description':
+                source_value = y
+            else:
+                continue
+        if source_name != "" and source_value != "":
+            externalSources_dict[source_name] = source_value
+        else:
+            print("error, externalsources has changed format in Alissa")
+    patient["annotation_sources"] = externalSources_dict  # add previous section to total patient info
+    
+    # add VUS info to patientdata
+    for item in variants:
+        variant = item
+        variant.update(patient)
+        db.insert_one(variant)
+    logger.info('finished uploading to mongodb: %s' % patient_dn_no)
+
+
+if __name__ == '__main__':
+    main()
+    end_time = datetime.now(timezone.utc).isoformat()
+    diff = (dateutil.parser.isoparse(start_time) - dateutil.parser.isoparse(end_time))
+    print("## \t\tFinished in %.2f minutes" % (diff.total_seconds()/60))
```

Comment:
It is common practice if you create a `main` function to add all code in `if __name__ == '__main__':` within that function.

Reply:
see other comment's answers.

Reply:
will make the main according to best practice and not have global variables